File: GymFitness/app/src/main/java/com/example/gymfitness/domain/models/Leaderboard.kt
package com.example.gymfitness.domain.models

import androidx.compose.ui.graphics.Color

data class LeaderboardEntry(
    val userId: String,
    val displayName: String,
    val avatarInitials: String,
    val avatarColor: Color,
    val weeklyPoints: Int,
    val workoutsThisWeek: Int,
    val currentStreak: Int,
    val isCurrentUser: Boolean,
    val steps: Int = 0
)


enum class LeaderboardPeriod { WEEKLY, MONTHLY, ALL_TIME }

sealed class LeaderboardUiState {
    object Loading : LeaderboardUiState()
    data class Success(
        val entries: List<LeaderboardEntry>,
        val period: LeaderboardPeriod,
        val currentUserEntry: LeaderboardEntry
    ) : LeaderboardUiState()
    data class Error(val message: String) : LeaderboardUiState()
}

data class FriendEntry(
    val userId: String,
    val displayName: String,
    val avatarInitials: String,
    val avatarColor: Color,
    val currentStreak: Int,
    val isOnline: Boolean,
    val code: String
)

sealed class FriendCodeUiState {
    object Loading : FriendCodeUiState()
    data class Success(
        val currentUserCode: String,
        val friends: List<FriendEntry>
    ) : FriendCodeUiState()
    data class Error(val message: String) : FriendCodeUiState()
}

File: Backend_Python/app/models/leaderboard.py
from pydantic import BaseModel, Field
from typing import Optional

class UserProfile(BaseModel):
    userId: str
    friendCode: str
    displayName: str
    avatarInitials: str

class WorkoutPoints(BaseModel):
    userId: str
    points: int
    period: str # "WEEKLY", "MONTHLY", "ALL_TIME"
    steps: Optional[int] = 0
    workoutsCount: Optional[int] = 0


class AddFriendRequest(BaseModel):
    userId: str
    friendCode: str

class LeaderboardEntry(BaseModel):
    userId: str
    friendCode: str
    displayName: str
    avatarInitials: str
    weeklyPoints: int = 0
    workoutsThisWeek: int = 0
    currentStreak: int = 0
    steps: int = 0

